grunttasks/xssmitigate.js

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var gumbo = require('gumbo-parser');
+var sets = require('simplesets');
+
+module.exports = function (grunt) {
+  grunt.registerTask('xss-flag', 'Flag potentially harmful files', [
+    // grunt addition for htmllint
+    'clean',
+    'selectconfig:dist',
+    'l10n-generate-pages',
+    'l10n-json-to-html',
+    'xss-parse'
+    ]
+  );
+
+  grunt.registerMultiTask('xss-parse', 'Parse HTML files to check for XSS vectors', function () {
+    /*eslint complexity: [2,7] */
+    // We can switch over to using the Set Object, when ES6 becomes ok to use.
+    var tagSet = new sets.Set(['a', 'span', 'strong', 'em']);
+    var attrSet = new sets.Set(['id', 'href', 'tabindex']);
+    this.files.forEach(function (file) {
+      var src = file.src[0];
+      var contents = grunt.file.read(src);
+      var tree = gumbo(contents, {
+        fragment: true
+      });
+      if (tree.childNodes && tree.childNodes !== []) {
+        var childNodes = tree.childNodes;
+        for (var i = 0; i < childNodes.length; i++) {
+          var child = childNodes[i];
+          if (child.nodeType === 1) {
+            if (tagSet.has(child.nodeName) && child.childNodes.length === 1) {
+              console.log('im a child', child.nodeName, 'with text', child.childNodes[0].textContent);
```

Comment:
**PROTIP:** When using `console.log()`(or `grunt.log.writeln()`) you can use `%s` and `%d` as placeholders, like this:

```js
console.log('im a child %s with text %s', child.nodeName, child.childNodes[0].textContent);
```